pkg/vclib/constants.go

```diff
@@ -56,5 +56,4 @@ const (
 	TestDefaultDatacenter = "DC0"
 	TestDefaultDatastore  = "LocalDS_0"
 	TestDefaultNetwork    = "VM Network"
-	testNameNotFound      = "enoent"
```

Comment:
This is used in tests:
```
-*- mode: grep; default-directory: "~/src/github.com/kubernetes/cloud-provider-vsphere/" -*-
Grep started at Fri Jul 27 14:58:25

git --no-pager grep -n -e testNameNotFound -- 
pkg/vclib/constants.go:59:	testNameNotFound      = "enoent"
pkg/vclib/datacenter_test.go:52:	_, err = GetDatacenter(ctx, vc, testNameNotFound)
pkg/vclib/datacenter_test.go:62:	_, err = dc.GetVMByUUID(ctx, testNameNotFound)
pkg/vclib/datacenter_test.go:72:	_, err = dc.GetVMByPath(ctx, testNameNotFound)
pkg/vclib/datacenter_test.go:82:	_, err = dc.GetDatastoreByPath(ctx, testNameNotFound) // invalid format
pkg/vclib/datacenter_test.go:88:		Datastore: testNameNotFound,
pkg/vclib/datacenter_test.go:89:		Path:      testNameNotFound,
pkg/vclib/datacenter_test.go:101:	_, err = dc.GetDatastoreByName(ctx, testNameNotFound)
pkg/vclib/datacenter_test.go:111:	_, err = dc.GetFolderByPath(ctx, testNameNotFound)
pkg/vclib/datacenter_test.go:126:	_, err = dc.GetVMMoList(ctx, []*VirtualMachine{vm}, []string{testNameNotFound}) // invalid property
pkg/vclib/datacenter_test.go:138:	_, err = dc.GetVirtualDiskPage83Data(ctx, diskPath+testNameNotFound)
pkg/vclib/datacenter_test.go:153:	_, err = dc.GetDatastoreMoList(ctx, []*Datastore{ds}, []string{testNameNotFound}) // invalid property
pkg/vclib/datacenter_test.go:164:		avm.Name: {testNameNotFound, diskPath},
pkg/vclib/datacenter_test.go:172:	if attached[avm.Name][testNameNotFound] {
pkg/vclib/utils_test.go:67:	_, err = finder.Datastore(ctx, testNameNotFound)

Grep finished with 15 matches found at Fri Jul 27 14:58:25
```

Reply:
Hi @dougm,

Good catch. I'm re-running the command with `--tests` enabled.

Reply:
Hi @dougm,

FWIW, the `--tests` flag does not prevent the false-positive:

```shell
$ gometalinter.v2 \
  --tests \
  --deadline=300s \
  --disable-all \
  --enable=deadcode \
  --enable=varcheck \
  --enable=structcheck \
  --enable=unparam \
  --enable=unused \
  $(for d in cluster cmd pkg; \
    do find "${d}" -type d; done | \
    sed 's~^~./~g' | tr '\n' ' ')
pkg/vclib/constants.go:55:1:warning: testNameNotFound is unused (deadcode)
pkg/cloudprovider/vsphere/instances.go:113:45:warning: parameter ctx is unused (unparam)
pkg/cloudprovider/vsphere/instances.go:113:66:warning: parameter user is unused (unparam)
pkg/cloudprovider/vsphere/instances.go:113:79:warning: parameter keyData is unused (unparam)
pkg/cloudprovider/vsphere/instances.go:119:37:warning: parameter ctx is unused (unparam)
pkg/cloudprovider/vsphere/instances.go:147:41:warning: parameter ctx is unused (unparam)
pkg/cloudprovider/vsphere/instances.go:339:47:warning: parameter ctx is unused (unparam)
pkg/vclib/pbm.go:88:74:warning: parameter dc is unused (unparam)
pkg/vclib/virtualmachine.go:413:143:warning: parameter result 0 (error) is never used (unparam)
```